File: backend/internal/handlers/gps.go
package handlers

import (
	"backend/internal/services"
	"backend/internal/utils"
	"database/sql"
	"fmt"
	"net/http"
)

//handler to request data from gps service
func GetGPSDataHandler(db *sql.DB) http.HandlerFunc {
	return func(w http.ResponseWriter, r *http.Request) {
		
		//invoke fetch gps service function
		data, err := services.FetchGPSData(db)
		if err != nil {
			utils.SendErrorResponse(w, "Failed to get GPS data", http.StatusInternalServerError)
			return
		}
		fmt.Printf("data: %v\n", data)
		//send back response
		utils.SendJSONResponse(w, data)
	}
}

File: frontend/src/stores/gps.js
import { defineStore } from 'pinia';
import api from '@/utils/axios';

export const useGPSStore = defineStore('gps', {
  state: () => ({
    gpsData: [],
    loading: false,
    error: null,
  }),

  actions: {
    async fetchGPSData() {
      this.loading = true;
      this.error = null;

      try {
        //fetch gps data
        const response = await api.get('/api/gps', {
        });
        this.gpsData = response.data;
      } catch (error) {
        console.error('Failed to fetch GPS data:', error);
        this.error = 'Failed to fetch GPS data';
      } finally {
        this.loading = false;
      }
    },
  },
});
